### ISSUE TITLE
Problem with \MT@maybe@textcmd

### ISSUE THREAD

**Issue by rallg** [schlcht/microtype#30]
Building some (not all) known-good documents may generate an error, stating that `\MT@maybe@textcmd` is undefined. If I persist with build, the resulting PDF looks OK.

Engine: lualatex. Texlive 2022, all up-to-date. Issue happened with MT 3.1, also 3.1a from here.

It seems that `\MT@maybe@textcmd` is defined within the `\leftprotrusion` macro. Does not seem to matter whether or not I turn protrusion off. I can fix the problem by placing this code immediately after `\begin{document}`:
```
\makeatletter
\let\MT@maybe@textcmd\@firstofone
\makeatother
```
Since the problem does not occur for all documents in my document class (novel), I cannot write a "minimal" document illustrating the issue. However, if you have `novel` with its docs, look in `doc/lualatex/novel/extras` for document `novel-testsuite.tex`. It built without problem two months ago, but generates the MT error now.

**Comment by schlcht** [comment]
As you say, `\MT@maybe@textcmd` is defined within `\leftprotrusion`, and it is *only* called by commands that `\leftprotrusion` itself calls, so I don't see how it could possibly be undefined.

Anyway, this is impossible to debug without a MWE (which is *always* possible to create) or at least more info: what is the exact error message? what is in the line the error message refers to? could you double-check whether microtype.sty and microtype-luatex.def have the same version? could you post the log file?

**Comment by schlcht** [comment]
PS: I've just compiled your `novel-testsuite.tex` and did not get any errors.

**Comment by rallg** [comment]
Ah. Thanks for compiling `novel-testsuite.tex`. Now I know that the problem does not lie with my document class or with that file, or with `microtype`. I do get the error with that file, even if I remove the images and other frills. Yet I do not get the error on some files using `novel`, and do not get the error with another custom document class that loads many fewer packages than does `novel`.

About two days ago, while updating via `tlmgr`, it told me to use a different repo. So I did. Everything now says it is up to date. My suspicion is that there may have been a cross-repo incompatibility in some other package. Evidently I had the bad luck to update just as TeX was freezing 2022. So, this look like I need to do a TeX system debug on my device.

Or, as they say in the world of TeX error messages, `Expected a number. You entered 0. If you continue, will substitute 0.`
In expl3 syntax, that is `__aw_foo_bar`.

I will shortly close this issue.

**Comment by rallg** [comment]
Today, I removed and reinstalled texlive 2022 (the parts I use) from scratch. Now all is good. I conclude that, as I guessed above, there was a repo sync problem when I updated. Whatever package was incorrect, did not emerge from its cage until it interacted with microtype, and only under some circumstances. Such is TeX.

Over at tex.stackexchange.com, user egreg looked at this yesterday, but could not discern anything. I removed that post.
